prime-router/src/main/kotlin/Report.kt

```diff
@@ -1511,25 +1511,29 @@ class Report : Logging {
             reportService: ReportService,
             metadata: Metadata? = null,
         ): String = if (header.receiver?.topic?.isSendOriginal == true) {
-                // the externalName of the root report should equal the submission payload name parameter
-                reportService.getRootReport(header.reportFile.reportId).externalName ?: formExternalFilename(
-                    header.reportFile.reportId,
-                    header.reportFile.schemaName,
-                    MimeFormat.valueOfIgnoreCase(header.reportFile.bodyFormat),
-                    header.reportFile.createdAt,
-                    metadata = metadata ?: Metadata.getInstance()
-                )
-            } else if (header.reportFile.externalName != null) {
-                header.reportFile.externalName
-            } else {
-                formExternalFilename(
-                    header.reportFile.reportId,
-                    header.reportFile.schemaName,
-                    MimeFormat.valueOfIgnoreCase(header.reportFile.bodyFormat),
-                    header.reportFile.createdAt,
-                    metadata = metadata ?: Metadata.getInstance()
-                )
-            }
+            // the externalName of the root report should equal the submission payload name parameter
+            reportService.getRootReport(header.reportFile.reportId).externalName ?: formExternalFilename(
+                header.reportFile.reportId,
+                header.reportFile.schemaName,
+                MimeFormat.valueOfIgnoreCase(header.reportFile.bodyFormat),
+                header.reportFile.createdAt,
+                metadata = metadata ?: Metadata.getInstance(),
+                translationConfig = header.receiver.translation,
+                nameFormat = header.receiver.translation.nameFormat
+            )
+        } else if (header.reportFile.externalName != null) {
+            header.reportFile.externalName
+        } else {
+            formExternalFilename(
+                header.reportFile.reportId,
+                header.reportFile.schemaName,
+                MimeFormat.valueOfIgnoreCase(header.reportFile.bodyFormat),
+                header.reportFile.createdAt,
+                metadata = metadata ?: Metadata.getInstance(),
+                translationConfig = header.receiver?.translation,
+                nameFormat = header.receiver?.translation?.nameFormat
+            )
+        }
```

Comment:
Do we need the `else if` and `else` blocks at all now? How should the outbound filename depend on the `isSendOriginal` setting?

Reply:
I think the logic could maybe get cleaned up here... but I'm hesitant to make additional adjustments for a bug fix.

I'm overall not sure when we want to return the `externalName`